openedx/features/enterprise_support/enrollments/tests/test_utils.py

```diff
@@ -181,4 +177,45 @@ def test_existing_enrollment_does_not_fail(
             enterprise_uuid=ENTERPRISE_UUID,
         )
 
-        mock_get_enrollment_api.assert_called_once()
+        assert mock_get_enrollment_api.call_count == 2
+
+    @mock.patch('openedx.features.enterprise_support.enrollments.utils.enrollment_api.update_enrollment')
+    @mock.patch('openedx.features.enterprise_support.enrollments.utils.enrollment_api.get_enrollment')
+    @mock.patch('openedx.features.enterprise_support.enrollments.utils.enrollment_api.add_enrollment')
+    @mock.patch('openedx.features.enterprise_support.enrollments.utils.User.objects.get')
+    @mock.patch('openedx.features.enterprise_support.enrollments.utils.transaction')
+    def test_upgrade_user_enrollment_mode(
+        self,
+        mock_tx,
+        mock_user_model,
+        mock_add_enrollment_api,
+        mock_get_enrollment_api,
+        mock_update_enrollment_api,
+    ):
+        mock_get_enrollment_api.return_value = {
+            'mode': 'audit',
+            'is_active': True,
+        }
+
+        mock_update_enrollment_api.return_value = {
+            'mode': 'verified',
+            'is_active': True,
+        }
```

Comment:
[curious] I wonder if we should have any test cases for the non-upgrade cases, e.g. where `mode` on the existing enrollment is already `verified`? That is, should we have an explicit assertion that the logic would keep it at `verified`.

Reply:
Added a test for that!